blockstore/splitstore/splitstore.go

```diff
@@ -737,6 +789,9 @@ func (s *SplitStore) protectTxnRefs(markSet MarkSet) error {
 			for c := range workch {
 				err := s.doTxnProtect(c, markSet)
 				if err != nil {
+					// I would _NOT_ proceed if we fail here. Any errors like
+					// this in GC should cause us to abort and (to avoid
+					// deleting anything).
```

Comment:
yeah, good point. will fix.

Reply:
Rewrote to use errgroup and return an error if it fails to protect.

... and yeah, errgroup is very neat indeed!